dvc/project.py

```diff
@@ -1165,24 +1169,22 @@ def metrics_add(self, path, typ=None, xpath=None):
     def metrics_remove(self, path):
         self._metrics_modify(path, delete=True)
 
-    def graph(self, from_directory=None):
+    def graph(self, stages=None, from_directory=None):
         import networkx as nx
         from dvc.exceptions import OutputDuplicationError
 
         G = nx.DiGraph()
         G_active = nx.DiGraph()
-        stages = self.stages(from_directory)
-
+        stages = stages or self.stages(from_directory)
         outs = []
-        outs_by_path = {}
+
         for stage in stages:
             for o in stage.outs:
-                existing = outs_by_path.get(o.path, None)
-                if existing is not None:
-                    stages = [o.stage.relpath, existing.stage.relpath]
-                    raise OutputDuplicationError(o.path, stages)
+                for out in outs:
+                    if o.path == out.path and o.stage.path != out.stage.path:
```

Comment:
I guess this `o.stage.path != out.stage.path` is the reason why OutputDuplicationError was not raised on `overwrite==True` as mentioned in the comment above :slightly_smiling_face:

Reply:
--overwrite is an explicit action, so we might as well just remove the existing dvcfile in dvc/stage.py where we check for it.